### PR TITLE
feat: generated impacts as a part of rule descriptions

### PR DESCRIPTION
**Pull request by jkodu**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message(s) follow our guidelines: https://github.com/dequelabs/axe-core/blob/develop/doc/code-submission-guidelines.md#git-commits
- [x] Changes to rules and checks appropriately support [Shadow DOM](https://github.com/dequelabs/axe-core/blob/develop/doc/developer-guide.md)
- [x] Changes have been tested in [major browsers and Assistive Technologies](https://github.com/dequelabs/axe-core/blob/develop/doc/accessibility-supported.md#accessibility-supported)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Description of the changes

- As a part of the documentation for rule descriptions, an additional column describing the impact/ severity of each rule has been added.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

@WilcoFiers - can you point me to the issue for the same?

### CODE REVIEW

**Comment by jkodu** [comment]
@WilcoFiers - review please. Amendment as requested in place.

File: build/configure.js [hunk 1]
```diff
@@ -145,19 +141,77 @@ function buildRules(grunt, options, commons, callback) {
 
 				return c.options === undefined ? id : c;
 			});
-
 		}
 
+		function parseImpactForRule(rule) {
+			
+
+			function capitalize(s) {
```
Comment (WilcoFiers) [inline build/configure.js]: I'm finding it difficult to review code that has nested functions like this. Github won't let me collapse this so I can't simply skip over them when reviewing the outer-most function. The point of abstracting is to let me skip over some of the details, but this way I have to consider multiple scopes when reading through this. I find this is harder to read than it even would have been if the abstractions weren't put in here at all. I'd much prefer that any function that can be moved to the top of the file be moved there.

@isner @stephendonner WDYT? Is that just me?

File: build/configure.js [hunk 2]
```diff
@@ -145,19 +141,77 @@ function buildRules(grunt, options, commons, callback) {
...
+				} else{
+					return [];
+				}
+			}
+
+			var cumulativeScores = getUniqueArr(
+				getScore(rule.any, true) 							// Any: get highest impact
+					.concat(getScore(rule.all, false)) 	// All: get all unique
+					.concat(getScore(rule.none, false)) // None: get all unique
+			).sort().reverse(); //order highest to lowest
```
Comment (WilcoFiers) [inline build/configure.js]: I would think lowest to highest. Not a big deal though.

File: build/configure.js (hunk 1 above)
Comment (stephenmathieson) [reply build/configure.js]: While I agree, this seems to be the existing code-style. We tend to nest function declarations a lot.

Don't think this should necessarily be a blocker, but maybe others feel differently. ¯\_(ツ)_/¯

File: build/configure.js [hunk 3]
```diff
@@ -145,19 +141,77 @@ function buildRules(grunt, options, commons, callback) {
...
+					}
+					return out;
+				}, []);
+			}
+
+			function getScore(checkCollection, onlyHighestScore) {
+				var scores = getImpactScores(checkCollection);
+				if(scores && scores.length) {
+					return onlyHighestScore
+						? [Math.max.apply(null, scores)]
```
Comment (stephenmathieson) [inline build/configure.js]: Might read better as:

```js
[Math.max(...scores)]
```

Comment (jkodu) [reply build/configure.js]: This file is not es6 friendly, leaving it as is...

File: build/configure.js (hunk 2 above)
Comment (jkodu) [reply build/configure.js]: Sorted - lowest to highest...

File: build/configure.js (hunk 1 above)
Comment (jkodu) [reply build/configure.js]: Ignoring above for now, until we reach a consensus...

File: build/configure.js [hunk 4]
```diff
@@ -145,19 +141,77 @@ function buildRules(grunt, options, commons, callback) {
...
+					return self.indexOf(value) === index;
+				})
+			}
+
+			function getImpactScores(checkCollection) {
+				return checkCollection.reduce(function (out, check) {
+					var id = typeof check === 'string' ? check : check.id;
+					var definition = clone(findCheck(checks, id));
+					if (!definition) {
+						grunt.log.error('check ' + id + ' not found');
```
[the remark this answers is not in the document]
Comment (jkodu) [reply build/configure.js]: Unsure, not a blocker so leaving as is, until I read up grunt docs.

File: build/configure.js [hunk 5]
```diff
@@ -145,19 +141,77 @@ function buildRules(grunt, options, commons, callback) {
...
+					return onlyHighestScore
+						? [Math.max.apply(null, scores)]
+						: getUniqueArr(scores);
+				} else{
+					return [];
+				}
+			}
+
+			var cumulativeScores = getUniqueArr(
+				getScore(rule.any, true) 							// Any: get highest impact
```
[the remark this answers is not in the document]
Comment (jkodu) [reply build/configure.js]: refactored so no more beautiful :shipit:

**Pull request merged**